config/config.go

```diff
@@ -122,6 +121,8 @@ func NewDefaultAgentConfig() *AgentConfig {
 		panic(err)
 	}
 
+	// Note: This only considers container sources that are already setup. It's possible that container sources may
+	//       need a few minutes to be ready.
```

Comment:
Since this is not trivial to fix, I've added a note about the caveat, and we can tackle it in the future.